### PR TITLE
Test/snapshot cdc

### PR DESCRIPTION
**Pull request by lianxmfor**
this PR is add snapshot + cdc integration test for oomcli

### CODE REVIEW

**Comment by wfxr** [comment]
sqlite integration tests failed at `test_stream_join.sh`

**Comment by lianxmfor** [comment]
> sqlite integration tests failed at `test_stream_join.sh`

good catch! I'll debug it locally.

**Comment by lianxmfor** [comment]
> > sqlite integration tests failed at `test_stream_join.sh`
> 
> good catch! I'll debug it locally.

fixed.

This is because the index of SQLite must be unique in a database, so only the first push succeeds, and all subsequent pushes fail because the index `idx` already exists.

**Comment by lianxmfor** [comment]
offline `mysql/postgres/sqlite/tidb` passed.

**Review by jinghancc — APPROVED** [review]
LGTM

File: internal/database/dbutil/schema.go [hunk 1]
```diff
@@ -68,9 +68,12 @@ func BuildIndexDDL(tableName string, indexName string, fields []string, backend
 	qt := QuoteFn(backend)
 	switch backend {
 	case types.BackendCassandra,
-		types.BackendMySQL,
-		types.BackendSQLite:
+		types.BackendMySQL:
 		return fmt.Sprintf("CREATE INDEX %s ON %s (%s)", qt(indexName), qt(tableName), qt(fields...))
```
Comment (jinghancc) [inline internal/database/dbutil/schema.go:72]: should we also make other database's index unique?

Comment (lianxmfor) [reply internal/database/dbutil/schema.go:72]: I also feel the same way, use a unique name here in oomstore to save one DB by one judgment.

Comment (lianxmfor) [reply internal/database/dbutil/schema.go]: done

**Pull request merged**